web/src/app/main/components/UserSettingsPopover.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import Button from '@mui/material/Button'
+import ButtonBase from '@mui/material/ButtonBase'
+import Grid from '@mui/material/Grid'
+import Popover from '@mui/material/Popover'
+import Typography from '@mui/material/Typography'
+import { OpenInNew, Logout } from '@mui/icons-material'
+import { useDispatch } from 'react-redux'
+import { authLogout } from '../../actions'
+import ThemePicker from './ThemePicker'
+import { CurrentUserAvatar } from '../../util/avatars'
+import AppLink from '../../util/AppLink'
+import { useConfigValue, useSessionInfo } from '../../util/RequireConfig'
+
+export default function UserSettingsPopover(): JSX.Element {
+  const [feedbackEnabled] = useConfigValue('Feedback.Enable')
+  const { userName } = useSessionInfo()
+  const firstName = userName?.split(' ')[0]
+  const dispatch = useDispatch()
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const logout = () => dispatch(authLogout(true))
+
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null)
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
+    setAnchorEl(event.currentTarget)
+  }
+
+  const handleClose = (): void => {
+    setAnchorEl(null)
+  }
```

Comment:
These don't do a lot and are only used once anyway -- can we just inline them to remove the indirection?